src/capi.cpp

```diff
@@ -1075,6 +1075,18 @@ int rtcAddTrackEx(int pc, const rtcTrackInit *init) {
 		             init->msid ? std::make_optional(string(init->msid)) : nullopt,
 		             init->trackId ? std::make_optional(string(init->trackId)) : nullopt);
 
+		auto currentRid = init->rids;
+		while (*currentRid != NULL) {
```

Comment:
In other places (for instance `iceServers`), the C API uses an additional `*Count` field for arrays instead of a NULL terminator, so you should do the same here for consistency.